### PR TITLE
FEATURE: BEND-56 Configure debugger

### PR DESCRIPTION
**Pull request by octapf**
Configure the internal debugger inside each controller, action, router.

- Eliminate Debugger singleton
- Implement first Debugger Log on one endpoint
- Change Constructor param 'headers' to optional
- Implement Debugger to App & middleware.mongoose
- Implement Debugger to all controllers and actions

### CODE REVIEW

File: src/http/actions/User.ts [hunk 1]
```diff
@@ -34,6 +38,11 @@ export module UserActions {
    * @param f {Function} (err?: [ErrorHandlerBody], dom?: Profile)
    */
   export function GetProfileAction(userId: string, f: Function) {
+    Debugger.shared._printLog(
+      DebbugerEnvironments.debugger,
+      DebuggerTypes.debug,
+      `GetProfileAction`
```
Comment (whitehatdevv) [inline src/http/actions/User.ts:44]: What is that??????

Comment (whitehatdevv) [reply src/http/actions/User.ts:44]: The debugger should print:  
```
`Action GetProfileAction with an userId ${userId}`
```

File: src/http/controller/User.ts [hunk 2]
```diff
@@ -47,6 +50,11 @@ export module UserController {
     res: express.Response,
     next: express.NextFunction
   ) => {
+    Debugger.shared._printLog(
+      DebbugerEnvironments.debugger,
+      DebuggerTypes.debug,
+      'GetProfile Controller'
```
Comment (whitehatdevv) [inline src/http/controller/User.ts:56]: Same as before, the debug message must be explained, please don't copy the hard name of the controller

**Review by whitehatdevv — CHANGES_REQUESTED** [review]

**Comment by octapf** [comment]
@whitehatdevv i will fix this now.. i left you a comment yesterday saying i'm not sure what to put on the debug log

**Comment by whitehatdevv** [comment]
> @whitehatdevv i will fix this now.. i left you a comment yesterday saying i'm not sure what to put on the debug log

Where is the comment?

**Review by Kehomaxd — DISMISSED** [review]

**Review by whitehatdevv — CHANGES_REQUESTED** [review]
Pls delete all the comments 
`Call to debugger`

It's really clear

**Review by PatyVilas — APPROVED** [review]

**Review by Kehomaxd — APPROVED** [review]

**Pull request merged**